querycheck/queryfilter/filter.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package queryfilter
+
+import (
+	"context"
+
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+// QueryFilter defines an interface for implementing declarative filtering logic to apply to query results before
+// the results are passed to a query check request.
+type QueryFilter interface {
+	Filter(context.Context, FilterQueryRequest, *FilterQueryResponse)
+}
+
+// FilterQueryRequest is a request to a filter function.
+type FilterQueryRequest struct {
+	// Query represents the parsed log messages relating to found resources returned by the `terraform query -json` command.
```

Comment:
nit: This just represents a single parsed log message
```suggestion
	// QueryItem represents a single parsed log message relating to a found resource returned by the `terraform query -json` command.
```